crates/solidity/outputs/rust/lib/src/generated/parsers.rs

```diff
@@ -0,0 +1,18353 @@
+                                                                    }
+                                                                } else {
+                                                                    Fail {
+                                                                        error: ParseError::new(
+                                                                            start, "Colon",
+                                                                        ),
+                                                                    }
+                                                                }
+                                                            } {
+                                                                Pass { node, error } => {
+                                                                    furthest_error =
+                                                                        error.map(|error| {
+                                                                            error.maybe_merge_with(
+                                                                                furthest_error,
+                                                                            )
+                                                                        });
+                                                                    node
+                                                                }
+                                                                Fail { error } => {
+                                                                    break Fail {
+                                                                        error: error
+                                                                            .maybe_merge_with(
+                                                                                furthest_error,
+                                                                            ),
+                                                                    }
+                                                                }
+                                                            };
+                                                            let result_1 = match {
+                                                                let start_position =
+                                                                    stream.position();
+                                                                match self.parse_expression(stream)
+                                                                {
+                                                                    Fail { error } => {
+                                                                        stream.set_position(
+                                                                            start_position,
+                                                                        );
+                               
```

Comment:
Not sure I understand why are we looping in every parser? IIUC, all of them only run once.

Reply:
`loop` is a way of doing an early break with a value. like a function with `return`

Reply:
Aha. Thanks.
It is unfortunate Rust doesn't support named blocks 😄  Otherwise, I can see users missing subtle cases where the loop might never break. Not a concern for the code here though.